### PR TITLE
Release Gateway 3.4.2.0

### PR DESCRIPTION
**Pull request by Guaris**
Update changelog for the Gateway 3.4.2.0 release

https://github.com/Kong/kong-ee/blob/next/3.4.x.x/changelog/3.4.2.0/3.4.2.0.md

https://konghq.atlassian.net/browse/DOCU-3563

### CODE REVIEW

File: app/gateway/changelog.md [hunk 1]
```diff
@@ -9,6 +9,39 @@ Changelog for supported Kong Gateway versions.
...
+  * Implemented a new grace period that lasts 30 days from the Kong Enterprise license expiration date. 
+    During the grace period all open source functionality will be available, and
+    Enterprise functionality will be set to read-only mode.
+  * Added support for counters such as routes, plugins, licenses, and deployment information to the license report.
+  * Added a checksum to the output of the license endpoint.
+### Fixes
+#### Core
+* DNS
+  * Fixed an issue with the DNS client was not adhering to configured timeouts in a predictable manner. [#11386](https://github.com/Kong/kong/pull/11386)
+  * Fixed an issue that caused the DNS client to resolve incorrectly during transient network and DNS server failures. [#11386](https://github.com/Kong/kong/pull/11386)
```
Comment (lena-larionova) [inline app/gateway/changelog.md:28]: These are the same issue with two different entries (see the link to the PR). You should combine them into one.

File: app/gateway/changelog.md [hunk 2]
```diff
@@ -9,6 +9,39 @@ Changelog for supported Kong Gateway versions.
...
+* The default value of the [`dns_no_sync`](/gateway/3.5.x/reference/configuration/#dns_no_sync) option has been changed to `on`.
+[#11871](https://github.com/kong/kong/pull/11871).
+#### Plugins
+* [**OpenTelemetry**](/hub/kong-inc/opentelemetry/) (`opentelemetry`): Fixed an issue that resulted in traces with invalid parent IDs when balancer instrumentation was enabled. [#11830](https://github.com/Kong/kong/pull/11830)
```
Comment (lena-larionova) [inline app/gateway/changelog.md:32]:
```suggestion
* [**OpenTelemetry**](/hub/kong-inc/opentelemetry/) (`opentelemetry`): 
  * Fixed an issue that resulted in traces with invalid parent IDs when balancer instrumentation was enabled. [#11830](https://github.com/Kong/kong/pull/11830)
```
I know this formatting seems excessive, but it's consistent with the rest of the changelog.

File: app/gateway/changelog.md [hunk 3]
```diff
@@ -9,6 +9,39 @@ Changelog for supported Kong Gateway versions.
...
+* [**OpenID Connect**](/hub/kong-inc/openid-connect/) (`openid-connect`): Fixed an issue with `using_pseudo_issuer`, where once propogated it was not used. [#6981](https://github.com/Kong/kong-ee/pull/6981)
```
Comment (lena-larionova) [inline app/gateway/changelog.md:33]:
```suggestion
* [**OpenID Connect**](/hub/kong-inc/openid-connect/) (`openid-connect`)
  * Fixed an issue with `using_pseudo_issuer`, where it was not used after it was propagated.
```
We can't add links to kong-ee in the changelog.

Not sure about this rephrase, you're welcome to adjust.

File: app/gateway/changelog.md [hunk 4]
```diff
@@ -9,6 +9,39 @@ Changelog for supported Kong Gateway versions.
...
+#### Kong Manager
```
Comment (lena-larionova) [inline app/gateway/changelog.md:34]: The Kong Manager section usually comes before Plugins.

File: app/gateway/changelog.md [hunk 5]
```diff
@@ -9,6 +9,39 @@ Changelog for supported Kong Gateway versions.
...
+#### Kong Manager 
+* Fixed an issue where some services were not showing the exact name or ID while configuring a route. 
+
+### Dependencies
+#### Enterprise
+* bump OpenSSL to 3.1.4
+* Add troubleshooting tools to container images 
+
+
+* Bumped `ngx_wasm_module` version to prerelease-0.1.1
```
Comment (lena-larionova) [inline app/gateway/changelog.md:43]:
```suggestion
* Bumped OpenSSL to 3.1.4
* Added troubleshooting tools to container images
* Bumped `ngx_wasm_module` version to prerelease-0.1.1 
```

Comment (lena-larionova) [inline app/gateway/changelog.md:43]: Bumped from what? Usually we do "bumped x from <some-version> to <new-version>".

File: app/gateway/changelog.md [hunk 6]
```diff
@@ -303,6 +303,40 @@ was called multiple times in a request lifecycle.
...
+* License management:
+  * Implemented a new grace period that lasts 30 days from the Kong Enterprise license expiration date. 
+    During the grace period all open source functionality will be available, and
+    Enterprise functionality will be set to read-only mode.
+  * Added support for counters such as routes, plugins, licenses, and deployment information to the license report.
+  * Added a checksum to the output of the license endpoint.
+### Fixes
+#### Core
+* DNS
+  * Fixed an issue with the DNS client was not adhering to configured timeouts in a predictable manner. Also fixed a related issue that cause the DNS client to resolve incorrectly during transient network and DNS server failures. [#11386](https://github.com/Kong/kong/pull/11386)
```
Comment (lena-larionova) [inline app/gateway/changelog.md:321]:
```suggestion
* Fixed an issue with the DNS client was not adhering to configured timeouts in a predictable manner. Also fixed a related issue that cause the DNS client to resolve incorrectly during transient network and DNS server failures. [#11386](https://github.com/Kong/kong/pull/11386)
```

**Review by lena-larionova — APPROVED** [review]

File: app/gateway/changelog.md [hunk 7]
```diff
@@ -303,6 +303,39 @@ was called multiple times in a request lifecycle.
...
+  * Implemented a new grace period that lasts 30 days from the Kong Enterprise license expiration date. 
+    During the grace period all open source functionality will be available, and
+    Enterprise functionality will be set to read-only mode.
+  * Added support for counters such as routes, plugins, licenses, and deployment information to the license report.
+  * Added a checksum to the output of the license endpoint.
+### Fixes
+#### Core
+* Fixed an issue with the DNS client was not adhering to configured timeouts in a predictable manner. Also fixed a related issue that cause the DNS client to resolve incorrectly during transient network and DNS server failures. [#11386](https://github.com/Kong/kong/pull/11386)
+* The default value of the [`dns_no_sync`](/gateway/3.5.x/reference/configuration/#dns_no_sync) option has been changed to `on`.
+[#11871](https://github.com/kong/kong/pull/11871).
```
Comment (cloudjumpercat) [inline app/gateway/changelog.md:322]:
```suggestion
[#11871](https://github.com/kong/kong/pull/11871).
* Removed a confusing log from the Redis rate limiting tool.
```

File: app/gateway/changelog.md [hunk 8]
```diff
@@ -303,6 +303,45 @@ was called multiple times in a request lifecycle.
   * Bumped `curl` from 8.3.0 to 8.4.0
   * Bumped `nghttp2` from 1.56.0 to 1.57.0
 
+## 3.4.2.0 
+**Release date** 2023/11/09
```
Comment (cloudjumpercat) [inline app/gateway/changelog.md:307]:
```suggestion
**Release date** 2023/11/10
```

**Pull request merged**